diff --git a/velox/connectors/pulsar/PulsarConnectorSplit.cpp b/velox/connectors/pulsar/PulsarConnectorSplit.cpp
index 2a494641f..10c92ab1b 100644
--- a/velox/connectors/pulsar/PulsarConnectorSplit.cpp
+++ b/velox/connectors/pulsar/PulsarConnectorSplit.cpp
@@ -31,6 +31,7 @@ folly::dynamic PulsarConnectorSplit::serialize() const {
   folly::dynamic obj = folly::dynamic::object;
   obj["connectorId"] = connectorId;
   obj["serviceUrl"] = serviceUrl_;
+  obj["name"] = "PulsarConnectorSplit";
   obj["topic"] = topic_;
   obj["subscriptionName"] = subscriptionName_;
   obj["format"] = format_;
